gossip/store_certificates.go

```diff
@@ -120,17 +125,40 @@ func getCertificate[S cert.Statement](
 	return res, res.Deserialize(data)
 }
 
+// getLatestCertificate retrieves the latest certificate from the key/value store.
+// If no certificate is found, an error is returned.
+func getLatestCertificate[S cert.Statement](
+	table kvdb.Store,
+) (cert.Certificate[S], error) {
+	var res cert.Certificate[S]
```

Comment:
nit: res is not used until the deserialization, I would rather it was declared just above it, and return explicit nils in error cases.

Reply:
As discussed offline, this version is deemed more readable then the alternative where we would have to define zero values in each early exit branch.